release-23.1: sql: remove stale skip in TestTelemetry

yuzefovich opened the pull request:
Backport #10755.

This commit unskips multiple telemetry tests that were skipped for no good reason (they were referencing an unrelated issue). This uncovered some bugs in the new schema changer telemetry reporting where we duplicated `_index` twice in the feature counter for inverted indexes. Also, `index` telemetry test contained an invalid statement which is now removed.

The only file that is still skipped is `sql-stats` where the output doesn't match the expectations, and I'm not sure whether the test is stale or something is broken, so a separate issue was filed.

Epic: None

Release note: None

Release justification: bug fix.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/107663)
<!-- Reviewable:end -->

yuzefovich committed:
sql: remove stale skip in TestTelemetry

This commit unskips multiple telemetry tests that were skipped for no
good reason (they were referencing an unrelated issue). This uncovered
some bugs in the new schema changer telemetry reporting where we
duplicated `_index` twice in the feature counter for inverted indexes.
Also, `index` telemetry test contained an invalid statement which is now
removed.

The only file that is still skipped is `sql-stats` where the output
doesn't match the expectations, and I'm not sure whether the test is
stale or something is broken, so a separate issue was filed.

Release note: None
